CMakeLists.txt

```
@@ -34,7 +34,10 @@ if("${CMAKE_SYSTEM_NAME}" STREQUAL "Emscripten")
         message(STATUS "Bundling game data from path: ${WEBASSEMBLY_GAME_PATH}")
     endif()
 
-    add_compile_options(-O3)
+    if("${CMAKE_BUILD_TYPE}" STREQUAL "Release")
+        # Emscripten's CMake toolchain defaults to O2, but we want O3
+        add_compile_options(-O3)
+    endif()
```

```suggestion
    # Emscripten's CMake toolchain defaults to O2, but we want O3
    add_compile_options($<$<NOT:$<CONFIG:Debug>>:-O3>)
```
So it also works with multi-configuration generators, and the `MinSizeRel` and `RelWithDebSymbols` configurations.

---

Nice! 👍